Horrible hack to make conky JUST compatible with the karmic kernel (2.6.31).
See LP #435571.

Index: conky-1.7.2/src/linux.c
===================================================================
--- conky-1.7.2.orig/src/linux.c	2009-09-24 15:33:48.000000000 +0200
+++ conky-1.7.2/src/linux.c	2009-09-24 15:37:11.000000000 +0200
@@ -858,25 +858,14 @@
 	}
 
 	if (strcmp(dir, "/sys/class/hwmon/") == 0) {
-		if (*buf) {
-			/* buf holds result from get_first_file_in_a_directory() above,
-			 * e.g. "hwmon0" -- append "/device" */
-			strcat(buf, "/device");
-		} else {
+	  if (!(*buf)) {
 			/* dev holds device number N as a string,
-			 * e.g. "0", -- convert to "hwmon0/device" */
-			sprintf(buf, "hwmon%s/device", dev);
+			 * e.g. "0", -- convert to "hwmon0" */
+			sprintf(buf, "hwmon%s", dev);
 			dev = buf;
 		}
 	}
 
-	/* At least the acpitz hwmon doesn't have a 'device' subdir,
-	 * so check it's existence and strip it from buf otherwise. */
-	snprintf(path, 255, "%s%s", dir, dev);
-	if (stat(path, &st)) {
-		buf[strlen(buf) - 7] = 0;
-	}
-
 	/* change vol to in, tempf to temp */
 	if (strcmp(type, "vol") == 0) {
 		type = "in";
